sdk/armcore/poller.go

```diff
@@ -0,0 +1,1068 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package armcore
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+const (
+	headerAsyncOperation = "Azure-AsyncOperation"
+	headerLocation       = "Location"
+)
+
+const (
+	operationInProgress string = "InProgress"
+	operationCanceled   string = "Canceled"
+	operationFailed     string = "Failed"
+	operationSucceeded  string = "Succeeded"
+)
+
+var pollingCodes = [...]int{http.StatusNoContent, http.StatusAccepted, http.StatusCreated, http.StatusOK}
+
+// CreatePollingTracker creates a polling tracker based on the verb of the original request and returns
+// the polling tracker implementation for the method verb or an error.
+// NOTE: this is only meant for internal use in generated code.
+func CreatePollingTracker(pollerType string, finalState string, resp *azcore.Response, errorHandler methodErrorHandler) (Poller, error) {
```

Comment:
I would rename this to `CreatePoller` as it returns a `Poller`.  The whole tracker thing is really an implementation detail.

Reply:
Done